### DISCUSSION TITLE
AnySoftKeyboard build error in Android studio!

### DISCUSSION THREAD

**Discussion by GoRaN9O** [AnySoftKeyboard/AnySoftKeyboard#2603] [Q&A] [answered]
Hello, 

How to fix this error in Parser.java when trying to build the project in Android Studio the master code is not import in latest version of Android studio and Java 11, Please help me how to fix that Error.

[![Screenshot-2020-12-21-185522.png](https://i.postimg.cc/y8KvxCgd/Screenshot-2020-12-21-185522.png)](https://postimg.cc/kBjxYZy3)

**Comment by alenappo** [comment]
i am trying to solve the same problem, can anyone help me?

**Reply by sslaia** [reply]
Hi @GoRaN909, @alenappo I can't help you with this (I only created the language addons), but as far as I know, there's no need to touch this particular file, so no idea how it was changed on your machine.

However as far as I can see from the screenshot, the ’ and ” character on your device are different from the original repository. You may try to fix it directly by copying and pasting the lines 169, 171 and 172 from the repository.

BTW I remember long time ago when I used Android Studio for compiling the addons, there would be a lot of error. Therefore I don't use AS or any IDE at all. I use a simple text editor for editing the xml files and compile the app on command line with **./gradlew -PwithAutoVersioning :addons:languages:LANGUAGE_NAME:apk:assemble** Hope this helps.

Edit: added the OP

**Reply by Taknok** [reply]
Could you past the snippet of code here ?

**Reply by alenappo** [reply]
I followed @sslaia  suggestion but it presented a new error:

[CXX1104] NDK from ndk.dir at G:\android\android-ndk-r14b had version [14.1.3816874] which disagrees with android.ndkVersion [23.0.7599858]
Affected Modules: jnidictionaryv1, jnidictionaryv2

the ndk version I installed is the one suggested here: https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/CONTRIBUTING.md

**Reply by GoRaN9O** [reply]
> Hi @GoRaN909, @alenappo I can't help you with this (I only created the language addons), but as far as I know, there's no need to touch this particular file, so no idea how it was changed on your machine.
> 
> However as far as I can see from the screenshot, the ’ and ” character on your device are different from the original repository. You may try to fix it directly by copying and pasting the lines 169, 171 and 172 from the repository.
> 
> BTW I remember long time ago when I used Android Studio for compiling the addons, there would be a lot of error. Therefore I don't use AS or any IDE at all. I use a simple text editor for editing the xml files and compile the app on command line with **./gradlew -PwithAutoVersioning :addons:languages:LANGUAGE_NAME:apk:assemble** Hope this helps.
> 
> Edit: added the OP

Hi and thank you for your suggestion I didn't touch any thing of original master code when i downloaded the ZIP file from GitHub source it's same with the file at github repository just extracted the files and tried to open with Android studio at the beginning the project is not sync with gradle and show that error attached on screenshot above!

**Comment by sslaia** [comment]
Remove the ndk.dir from the local.properties file


On Wed, 11 May 2022 at 17:28, Alessia ***@***.***> wrote:

> I followed @sslaia <https://github.com/sslaia> suggestion but it
> presented a new error:
>
> [CXX1104] NDK from ndk.dir at G:\android\android-ndk-r14b had version
> [14.1.3816874] which disagrees with android.ndkVersion [23.0.7599858]
> Affected Modules: jnidictionaryv1, jnidictionaryv2
>
> the ndk version I installed is the one suggested here:
> https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/CONTRIBUTING.md
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/AnySoftKeyboard/AnySoftKeyboard/discussions/2603#discussioncomment-2731768>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AKWQYAQJGQEJS52WNNMFKNTVJPNZFANCNFSM4VEMYSMA>
> .
> You are receiving this because you were mentioned.Message ID:
> <AnySoftKeyboard/AnySoftKeyboard/repo-discussions/2603/comments/2731768@
> github.com>
>
-- 

Sirus

**Comment by menny** [comment, marked as answer]
Something is changing those files on your local file system. If you look at those lines in github web: https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/main/buildSrc/src/main/java/Parser.java#L169 you will notice that these are *different* characters.
The image you shared shows:
```
switch (intChar) {
            case ''':
                return '\'';
            case '"':
            case '"':
                return '\"';
            default:
                return (char) intChar;
        }
```

while it should be:
```
switch (intChar) {
            case '’':
                return '\'';
            case '”':
            case '“':
                return '\"';
            default:
                return (char) intChar;
        }
```

**Reply by menny** [reply]
To help out with this, I created https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/3356

**Comment by GoRaN9O** [comment]
Thank you so much Mr. @menny I can confirm that the Error is fixed after those changes you made on this PR! #3356
